pkg/controller/argocd/util.go

```diff
@@ -1067,7 +1070,82 @@ func namespaceFilterPredicate() predicate.Predicate {
 			if _, ok := e.MetaNew.GetLabels()[common.ArgoCDManagedByLabel]; ok {
 				return true
 			}
+			if ns, ok := e.MetaOld.GetLabels()[common.ArgoCDManagedByLabel]; ok && ns != ""{
+				if err := deleteRBACsForNamespace(ns, e.MetaOld.GetName()); err != nil {
+					log.Error(err, fmt.Sprintf("failed to delete RBACs for namespace: %s", e.MetaOld.GetName()))
+				}
+				log.Info(fmt.Sprintf("Successfully removed the RBACs for namespace: %s", e.MetaOld.GetName()))
+			}
 			return false
 		},
 	}
 }
+
+func deleteRBACsForNamespace(ownerNS, sourceNS string) error {
+	log.Info(fmt.Sprintf("Removing the RBACs created for the namespace: %s", sourceNS))
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		log.Error(err, fmt.Sprintf("unable to get k8s config"))
+		return err
+	}
```

Comment:
Will this work when the Operator is run locally? We had that issue before I think.

Reply:
Thanks Jann!! Completely forgot about that. Fixed it!! Also added comments